.github/workflows/proto-sync.yml

```diff
@@ -0,0 +1,72 @@
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'packages/core/protocols/src/proto/**'
+  workflow_dispatch:
+
+name: sync-proto
+
+jobs:
+  copy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Get current commit
+        run: echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+
+      - name: Get new branch name
+        run: echo "BRANCH_NAME=${{ env.BRANCH_NAME_PREFIX }}-${{ env.COMMIT_HASH }}" >> $GITHUB_ENV
+        env:
+          BRANCH_NAME_PREFIX: proto-sync
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+
+      - name: Install protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install protoc deps
+        run: |
+          go install github.com/golang/protobuf/protoc-gen-go@latest
+          go install github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc@latest
+
+      - name: Update KUBE repo
+        run: |
+          # Clone the repo.
+          rm -rf target
+          git clone https://.:${{ secrets.CREATE_PR_TOKEN }}@github.com/dxos/kube target
+          cd target
+
+          # Prepare tools.
+          go build ./cmds/protoc-gen-go-dxrpc
+          sudo mv protoc-gen-go-dxrpc /usr/local/bin/protoc-gen-go-dxrpc
+
+          # Update protos & gen defs.
+          cp -r ../packages/core/protocols/src/proto proto
+          go generate ./...
+
+          # Setup Git.
+          git config user.name "dxos-bot"
+          git config user.email "bot@dxos.org"
+
+          # Setup branch.
+          git checkout -B ${{ env.BRANCH_NAME }}
+          git add .
+          git diff-index --quiet HEAD || git commit -m "fix: automatic .proto clone from dxos repo"
+          git push origin ${{ env.BRANCH_NAME }}
+
+          # Open Pull Request.
+          gh pr create -B main -H ${{ env.BRANCH_NAME }} --title '${{ env.PR_TITLE }} from ${{ env.COMMIT_HASH }}' --body '${{ env.PR_BODY }}'
+        env:
+          GH_TOKEN: ${{ secrets.CREATE_PR_TOKEN }}
+          PR_TITLE: "fix: sync protocol buffers"
```

Comment:
I usually use `chore` for any bot things, then it doesn't pollute the change logs